packages/node/src/integrations/anr/worker.ts

```diff
@@ -0,0 +1,214 @@
+    dist: options.dist,
+    platform: 'node',
+    level: 'error',
+    exception: {
+      values: [
+        {
+          type: 'ApplicationNotResponding',
+          value: `Application Not Responding for at least ${options.anrThreshold} ms`,
+          stacktrace: { frames },
+          // This ensures the UI doesn't say 'Crashed in' for the stack trace
+          mechanism: { type: 'ANR' },
+        },
+      ],
+    },
+    tags: { 'process.name': 'ANR' },
+  };
+
+  if (options.debug) {
+    // eslint-disable-next-line no-console
+    console.log(JSON.stringify(event));
+  }
+
+  const envelope = createEventEnvelope(event, options.dsn, options.sdkMetadata);
+  await transport.send(envelope);
+  await transport.flush(2000);
+
+  // Delay for 5 seconds so that stdio can flush in the main event loop ever restarts.
+  // This is mainly for the benefit of logging/debugging issues.
+  setTimeout(() => {
+    process.exit(0);
+  }, 5_000);
+}
+
+let debuggerPause: () => void | undefined;
+
+if (options.captureStackTrace) {
+  log('Connecting to debugger');
+
+  const session = new InspectorSession() as InspectorSessionNodeV12;
+  session.connectToMainThread();
+
+  log('Connected to debugger');
+
+  // Collect scriptId -> url map so we can look up the filenames later
+  const scripts = new Map<string, string>();
+
+  session.on('Debugger.scriptParsed', event => {
+    scripts.set(event.params.scriptId, event.params.url);
+  });
+
+  session.on('Debugger.paused', event => {
+    if (event.params.reason !== 'other') {
+      return;
+    }
+
+    try {
+      log('Debugger paused');
+
+      // copy the frames
+      const callFrames = [...event.params.callFrames];
+
+      const stackFrames = stripSentryFramesAndReverse(
+        callFrames.map(frame => callFrameToStackFrame(frame, scripts.get(frame.location.scriptId), () => undefined)),
+      );
+
+      // Evaluate a script in the currently paused context
+      session.post(
+        'Runtime.evaluate',
+        {
+          // Grab the trace context from the current scope
+          expression: 'const ctx = __SENTRY__.hub.getScope().getPropagationContext(); ctx.traceId + "-" + ctx.spanId',
+          // Don't re-trigger the debugger if this causes an error
+          silent: true,
+        },
+        (_, param) => {
+          const traceId = param && param.result ? (param.result.value as string) : '-';
+          const [trace_id, span_id] = traceId.split('-');
+
+          session.post('Debugger.resume');
+          session.post('Debugger.disable');
+
+          const context = trace_id && span_id ? { trace_id, span_id } : undefined;
+          void sendAnrEvent(stackFrames, context);
+        },
+      );
+    } catch (e) {
+      session.post('Debugger.resume');
+      session.post('Debugger.disable');
+      throw e;
+    }
+  });
+
+  debuggerPause = () => {
+    try {
+      session.post('Debugger.enable', () => {
+        session.post('Debugger.pause');
+      });
+    } catch (_) {
```

Comment:
Maybe we should just add a linter rule for this...

```suggestion
    } catch {
```

Reply:
We need to ensure our tooling reinserts it because this was only introduced in ECMAScript 2019.